### PR TITLE
feat(naive_bayes): add CategoricalNB classifier

### LINKED ISSUE #1399: CategoricalNB
- [https://scikit-learn.org/stable/modules/generated/sklearn.naive_bayes.CategoricalNB.html](https://scikit-learn.org/stable/modules/generated/sklearn.naive_bayes.CategoricalNB.html)
- [https://github.com/online-ml/river/discussions/774](https://github.com/online-ml/river/discussions/774)

### PR DESCRIPTION
**Pull request by uttam12331**
Closes #1399.

Adds an online `CategoricalNB` naive Bayes classifier for **categorical features** (e.g. `{"weather": "sunny", "humidity": "high"}`), mirroring scikit-learn's [`CategoricalNB`](https://scikit-learn.org/stable/modules/generated/sklearn.naive_bayes.CategoricalNB.html) but learning incrementally. It fills the gap alongside the existing `MultinomialNB`/`BernoulliNB`/`ComplementNB`/`GaussianNB`.

### Model
For each feature `f`, category `v` and class `c`:

```
P(x_f = v | c) = (count[f][(c, v)] + alpha) / (class_count[c] + alpha * n_categories(f))
```

with the empirical class prior `P(c) = class_count[c] / N`. New categories seen after the first observations are handled gracefully (online setting).

### Implementation
- `learn_one` / `learn_many` (mini-batch via `MiniBatchClassifier`), `joint_log_likelihood` / `joint_log_likelihood_many`, plus `p_class` / `p_feature_given_class` helpers — consistent with the other NB classes.
- Registered in `naive_bayes/__init__.py`.

### Verification
- **Matches scikit-learn's `CategoricalNB` to machine precision** (max `predict_proba` difference `~4e-16` across alphas) — covered by a new `test_categorical_vs_sklearn` (parametrized over alpha).
- `learn_many` produces an identical model to repeated `learn_one` (`test_categorical_learn_many_vs_learn_one`).
- Unseen categories at predict time don't raise and keep probabilities normalized (`test_categorical_handles_unseen_feature_value`).
- A runnable doctest example in the class docstring.
- Full `river/naive_bayes` suite passes (50 tests), the **generic estimator checks pass** (`test_estimators.py -k CategoricalNB`, 53 checks), and `ruff check` / `ruff format` are clean.

### CODE REVIEW

File: river/naive_bayes/categorical.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        """
+        if not self.class_counts:
+            return {}
+        return {
+            c: math.log(self.p_class(c))
+            + sum(math.log(self.p_feature_given_class(f, value, c)) for f, value in x.items())
+            for c in self.classes_
+        }
+
+    def learn_many(self, X: pd.DataFrame, y: pd.Series):
```
Comment (MaxHalford) [inline river/naive_bayes/categorical.py:127]: Could you support Narwhals? We're in the process of moving all mini-batch methods to Narwhals instead of pandas

File: river/naive_bayes/categorical.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        Parameters
+        ----------
+        X
+            A dataframe of categorical feature values.
+        y
+            A series of target classes.
+
+        """
+        for (_, row), label in zip(X.iterrows(), y):
+            self.learn_one(row.to_dict(), label)
```
Comment (MaxHalford) [inline river/naive_bayes/categorical.py:139]: We do not want mini-batch methods to be for loops over the inputs. Mini-batch methods must use vectorization, else they're not bringing anything to the table.

**Review by MaxHalford — COMMENTED** [review]
Nice addition!
